### DISCUSSION TITLE
Plans for graph traversal and filtering predicates

### DISCUSSION THREAD

**Discussion by rvlasveld** [lolopinto/ent#1002] [Ideas]
I was wondering what the plans are around graph traversal and filtering. 

Currently we have EntQueries, which allows constructs like `user.queryFriends()`. We also have `first()` and `last()`, to limit the number of results. 

To use filtering, it seems that we need to use custom queries in the form of `User.loadCustom()`, where a string or Clause can be passed. 

It's not clear to me what's the best way to combine traversal and filtering. Looking at FB's EntSchema, or other implementations (such as [EntGo](https://entgo.io/docs/predicates)), there are APIs like the following: `user.queryFriends().whereName(predicate.equals("John")).queryHomeTown()`, etc. 

Are there plans for this type of graph traversal and predicate-based filtering as a single API?

**Comment by lolopinto** [comment]
eventually, yes. just hasn't been prioritized 

https://github.com/lolopinto/ent/issues/685 exists as high level task tracking this
